src/integrationTest/java/com/mongodb/spark/sql/connector/beans/DateTimeBean.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.mongodb.spark.sql.connector.beans;
+
+import java.io.Serializable;
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.Objects;
+
+public class DateTimeBean implements Serializable {
+  private java.sql.Date sqlDate;
+  private java.sql.Timestamp sqlTimestamp;
+  private java.time.LocalDate localDate;
+  private java.time.Instant instant;
+
+  public DateTimeBean() {}
+
+  public DateTimeBean(
+      final Date sqlDate,
+      final Timestamp sqlTimestamp,
+      final LocalDate localDate,
+      final Instant instant) {
+    this.sqlDate = sqlDate;
+    this.sqlTimestamp = sqlTimestamp;
+    this.localDate = localDate;
+    this.instant = instant;
+  }
+
+  public Date getSqlDate() {
+    return sqlDate;
+  }
+
+  public void setSqlDate(final Date sqlDate) {
+    this.sqlDate = sqlDate;
+  }
+
+  public Timestamp getSqlTimestamp() {
+    return sqlTimestamp;
+  }
+
+  public void setSqlTimestamp(final Timestamp sqlTimestamp) {
+    this.sqlTimestamp = sqlTimestamp;
+  }
+
+  public LocalDate getLocalDate() {
+    return localDate;
+  }
+
+  public void setLocalDate(final LocalDate localDate) {
+    this.localDate = localDate;
+  }
+
+  public Instant getInstant() {
+    return instant;
+  }
+
+  public void setInstant(final Instant instant) {
+    this.instant = instant;
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    DateTimeBean that = (DateTimeBean) o;
+    // Cheat with sqlDate - ignore system clock normalization
+    return Objects.equals(sqlDate.toString(), that.sqlDate.toString())
```

Comment:
Spent so much time playing with this shit, and I still can't understand why is it OK for us to read back `java.sql.Date` that is not `equal` to the one we wrote?

Reply:
The string value is the same. But I found when hydrated through spark it normalizes the value which causes the issue. So I opted to cheat and just use the string unnormalized representation.

Reply:
Fixed by temp setting the timezone of the machine to UTC for the test.